nodes/obstacle_visualizer.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python3
+
+from typing import List, Dict
+
+import rospy
+
+from std_msgs.msg import Header
+from geometry_msgs.msg import Pose
+from tf.transformations import quaternion_from_euler
+from geometry_msgs.msg import Point, Quaternion, Vector3
+from visualization_msgs.msg import Marker, MarkerArray
+from std_msgs.msg import ColorRGBA, Header
+from SDFGenerator import SDFGenerator
+
+from gazebo_msgs.srv import SpawnModel, SpawnModelRequest
+
+
+class ObstaclesVisualizer:
+    def __init__(self, name: str) -> None:
+        self._node_name = name
+        rospy.init_node(self._node_name, anonymous=False)
+
+        obstacle_idx = 1
+
+        if not rospy.has_param("~obstacle1"):
+            rospy.logerr(
+                f"[{rospy.get_name()}] Param obstacle1 not found. No collision avoidance with an external obstacle computed!"
+            )        
+
+        self._obstacles_infos = {}
+        self._obstacles = []
+        while rospy.has_param("~obstacle" + str(obstacle_idx)):
+            obstacle_name = "~obstacle" + str(obstacle_idx)
+            self._obstacles_infos.update(
+                {obstacle_name: rospy.get_param(obstacle_name)}
+            )
+            obstacle_idx += 1
+
+        sdf_generator = SDFGenerator()
+        self._spawn_model_requests = []
+        self._markers = MarkerArray()
+        header = Header(frame_id="world", stamp=rospy.Time.now())
+        for iter, (key, obstacle) in enumerate(self._obstacles_infos.items()):
+            
+            pose=self._parse_poses(obstacle)
+ 
+            sp_req = SpawnModelRequest(
+                model_name = key[1:],
+                robot_namespace = "",
+                initial_pose = pose,
+                reference_frame = header.frame_id
+            )
+            
+
+            m = Marker(
+                header=header,
+                ns="",
+                id=iter,
+                action=Marker.MODIFY,
+                pose = pose,
+                color=ColorRGBA(r=0.58, g=0.0, b=0.98, a=0.7),
+                lifetime=rospy.Duration(0.1),
+                frame_locked=False,
+            )
+            if obstacle["type"] == "sphere":
+                m.scale = Vector3(**dict(zip("xyz", [obstacle["radius"] * 2] * 3)))
```

Comment:
I know you hate this line as much as I love it so feel free to replace it with
```python
d = obstacle["radius"] * 
m.scale = Vector3(x=d, y=d, z=d)
```
Or something similar

Reply:
it's fine let's keep the complicated things here so when ppl see this they'll think we're smart

Reply:
Your're smart*